src/azure-cli/azure/cli/command_modules/resource/_validators.py

```diff
@@ -32,9 +74,13 @@ def _validate_deployment_name(namespace):
 
 
 def process_deployment_create_namespace(namespace):
-    if bool(namespace.template_uri) == bool(namespace.template_file):
-        raise CLIError('incorrect usage: --template-file FILE | --template-uri URI')
-    _validate_deployment_name(namespace)
+    if bool(namespace.template_uri) == bool(namespace.template_file) == bool(namespace.template_spec):
```

Comment:
It seems this logic is not correct.
If bool(namespace.template_uri) is True, bool(namespace.template_file) is False, and bool(namespace.template_spec) is True, no error is raised. Actually, it should raise error in that case.

You can use "if [bool(namespace.template_uri), bool(namespace.template_file), bool(namespace.template_spec)].count(True) > 1:" as condition check.

Reply:
When will it throw exception in this method?

---
In reply to: [465502144](https://github.com/Azure/azure-cli/pull/14448#discussion_r465502144) [](ancestors = 465502144)